src/libs/UserUtils.ts

```diff
@@ -115,64 +103,57 @@ function getDefaultAvatarURL(accountID = '', isNewDot = false) {
-        _.isString(avatarURL) &&
-        (avatarURL.includes('images/avatars/avatar_') || avatarURL.includes('images/avatars/default-avatar_') || avatarURL.includes('images/avatars/user/default'))
-    ) {
-        return true;
-    }
-
-    // We use a hardcoded "default" Concierge avatar
-    if (_.isString(avatarURL) && (avatarURL === CONST.CONCIERGE_ICON_URL_2021 || avatarURL === CONST.CONCIERGE_ICON_URL)) {
-        return true;
+function isDefaultAvatar(avatarURL: string): boolean {
+    if (typeof avatarURL === 'string') {
+        if (avatarURL.includes('images/avatars/avatar_') || avatarURL.includes('images/avatars/default-avatar_') || avatarURL.includes('images/avatars/user/default')) {
+            return true;
+        }
+
+        // We use a hardcoded "default" Concierge avatar
+        if (avatarURL === CONST.CONCIERGE_ICON_URL_2021 || avatarURL === CONST.CONCIERGE_ICON_URL) {
+            return true;
+        }
     }
 
     if (!avatarURL) {
         // If null URL, we should also use a default avatar
         return true;
     }
+
     return false;
 }
 
 /**
  * Provided a source URL, if source is a default avatar, return the associated SVG.
  * Otherwise, return the URL pointing to a user-uploaded avatar.
  *
- * @param {String} avatarURL - the avatar source from user's personalDetails
- * @param {Number} accountID - the accountID of the user
- * @returns {String|Function}
+ * @param avatarURL - the avatar source from user's personalDetails
+ * @param accountID - the accountID of the user
  */
-function getAvatar(avatarURL, accountID) {
+function getAvatar(avatarURL: string, accountID: number): React.FC<SvgProps> | string {
     return isDefaultAvatar(avatarURL) ? getDefaultAvatar(accountID) : avatarURL;
 }
 
 /**
  * Provided an avatar URL, if avatar is a default avatar, return NewDot default avatar URL.
  * Otherwise, return the URL pointing to a user-uploaded avatar.
  *
- * @param {String} avatarURL - the avatar source from user's personalDetails
- * @param {Number} accountID - the accountID of the user
- * @returns {String}
+ * @param avatarURL - the avatar source from user's personalDetails
+ * @param accountID - the accountID of the user
  */
-function getAvatarUrl(avatarURL, accountID) {
+function getAvatarUrl(avatarURL: string, accountID: number): string {
     return isDefaultAvatar(avatarURL) ? getDefaultAvatarURL(accountID, true) : avatarURL;
 }
 
 /**
  * Avatars uploaded by users will have a _128 appended so that the asset server returns a small version.
  * This removes that part of the URL so the full version of the image can load.
- *
- * @param {String} [avatarURL]
- * @param {Number} [accountID]
- * @returns {String|Function}
  */
-function getFullSizeAvatar(avatarURL, accountID) {
+function getFullSizeAvatar(avatarURL: string, accountID: number): React.FC<SvgProps> | string {
     const source = getAvatar(avatarURL, accountID);
-    if (!_.isString(source)) {
+    if (typeof source !== 'string') {
```

Comment:
same question as above

Reply:
source is of type `React.FC<SvgProps> | string` so the condition here seems to be valid

Reply:
True, I think I didn't read it correctly at the time.